src/main/java/com/lists/CircularLinkedList.java

```diff
@@ -0,0 +1,215 @@
+
+    /*
+        checks is the
+        list is empty
+     */
+
+    public boolean isEmpty(){
+        return size == 0;
+    }
+
+    /*
+        add item to the last
+        of the list
+     */
+
+    public void add(Item item){
+        LinkedListNode linkedListNode = new LinkedListNode(item);
+        if(isEmpty()){
+            head = linkedListNode;
+            tail = linkedListNode;
+        }
+        else{
+            tail.next = linkedListNode;
+            tail = linkedListNode;
+
+        }
+        linkedListNode.next = head;
+        size++;
+    }
+
+    /*
+        add item at the
+        specified index
+     */
+
+    public void add(Item item, int index)throws IndexOutOfBoundsException{
+        if(index < 0 || index > size - 1){
+            throw new IndexOutOfBoundsException();
+        }
+
+        LinkedListNode linkedListNode = new LinkedListNode(item);
+
+        if(index == size - 1){
+            add(item);
+        }
+
+        else if(index == 0){
+            linkedListNode.next = head.next;
+            head = linkedListNode;
+            size++;
+        }
+
+        else {
+            LinkedListNode start = head;
+            while (index != 1) {
+                index--;
+                start = start.next;
+            }
+
+            linkedListNode.next = start.next;
+            start.next = linkedListNode;
+            size++;
+        }
+
+    }
+
+    /*
+        returns the item
+        at the specified index
+     */
+
+    public Item get(int index) throws IndexOutOfBoundsException{
+        if(index < 0 || index > size - 1){
+            throw new IndexOutOfBoundsException();
+        }
+
+        if(index == 0){
+            return head.item;
+        }
+
+        else if(index == size - 1){
+            return tail.item;
+        }
+
+        else{
+            LinkedListNode start = head;
+            while (index != 0) {
+                index--;
+                start = start.next;
+            }
+
+            return start.item;
+        }
+    }
+
+    /*
+        remove the item at the
+        specified index
+     */
+
+    public void remove(int index) throws IndexOutOfBoundsException{
+        if(index < 0 || index > size - 1){
+            throw new IndexOutOfBoundsException();
+        }
+
+        if(index == 0){
+            head = head.next;
+            tail.next = head;
+        }
+
+        else{
+            LinkedListNode start = head;
+            int temp = index;
+            while(temp != 1){
+                temp--;
+                start = start.next;
+            }
+
+            start.next = start.next.next;
+
+            if(index == size - 1){
+                tail = start;
+            }
+
+            size--;
+        }
+    }
+
+    /*
+        returns an iterator
+        to list
+     */
+
+    public Iterator<Item> iterator(){
+        return new CircularLinkedListIterator();
+    }
+
+    class CircularLinkedListIterator implements Iterator<Item> {
```

Comment:
Also make all inner classes inside this class "private". We do not want to others to access these classes at all.